dissect/target/loaders/mqtt.py

```diff
@@ -62,12 +67,79 @@ def _read(self, offset: int, length: int, optimization_strategy: int = 0) -> byt
         return data
 
 
+class MQTTDiagnosticLine:
+    def __init__(self, connection: MQTTConnection, total_peers: int):
+        self.connection = connection
+        self.total_peers = total_peers
+        self._columns, self._rows = os.get_terminal_size(0)
+        atexit.register(self._detach)
+        self._attach()
+
+    def _attach(self) -> None:
+        sys.stderr.write(f"\0337\033[r\0338\033D\033M\0337\033[1;{self._rows - 1}r\0338")
+
+    def _detach(self) -> None:
+        sys.stderr.write(f"\0337\033[{self._rows};1H\033[K\033[r\0338")
+        sys.stderr.flush()
+
+    def display(self) -> None:
+        prefix = "\x1b[44m\x1b[37m\r"
+        suffix = "\x1b[0m"
+        separator = "\x1b[41m\x1b[1m"
+        logo = "TARGETD"
+        start = time.time()
+        t2 = start
+        mark = start
+        _bytes = 0
+        subtract = 0
+        while True:
+            time.sleep(0.05)
+            peers = "?"
+            try:
+                peers = len(self.connection.broker.peers(self.connection.host))
+            except Exception:
+                pass
+            total = self.total_peers
+            recv = self.connection.broker.bytes_received
+            now = time.time()
+
+            # to avoid endless countdowns if no data is transferred in reality anymore.
+            if (now - mark) > 3 and not _bytes:
+                _bytes = recv
+                t2 = now
+            if (now - mark) > 9 and _bytes:
+                subtract = _bytes
+                _bytes = 0
+                mark = t2
+
+            recv -= subtract
+            failures = self.connection.retries
+            transfer = (recv / (now - mark)) / 1000
+            seconds_elapsed = round(now - start) % 60
+            minutes_elapsed = math.floor(seconds_elapsed / 60) % 60
+            hours_elapsed = math.floor(minutes_elapsed / 60)
+            timer = f"{hours_elapsed:02d}:{minutes_elapsed:02d}:{seconds_elapsed:02d}"
+            display = f"{timer} {peers}/{total} peers {transfer:>8.2f} KB p/s {failures:>4} failures"
+            rest = self._columns - len(display)
+            padding = (rest - len(logo) - 1) * " "
+            sys.stderr.write(f"\0337\033[{self._rows};1H\033[?7l\033[0m")
+            sys.stderr.write(prefix + display + padding + separator + logo + suffix)
+            sys.stderr.write("\033[?7h\0338")
+            sys.stderr.flush()
```

Comment:
```suggestion
            recv -= subtract
            failures = self.connection.retries
            transfer = (recv / (now - mark)) / 1000

            seconds_elapsed = round(now - start) % 60
            minutes_elapsed = math.floor(seconds_elapsed / 60) % 60
            hours_elapsed = math.floor(minutes_elapsed / 60)
            timer = f"{hours_elapsed:02d}:{minutes_elapsed:02d}:{seconds_elapsed:02d}"

            display = f"{timer} {peers}/{total} peers {transfer:>8.2f} KB p/s {failures:>4} failures"
            rest = self._columns - len(display)
            padding = (rest - len(logo) - 1) * " "

            sys.stderr.write(f"\0337\033[{self._rows};1H\033[?7l\033[0m")
            sys.stderr.write(prefix + display + padding + separator + logo + suffix)
            sys.stderr.write("\033[?7h\0338")
            sys.stderr.flush()
```
Some new lines to improve readability.